### PR TITLE
feat: new filter UI

### PR DESCRIPTION
**Pull request by jamie-rasmussen**
Internal weave: https://wandb.atlassian.net/browse/WB-19861

Adds a new UI for filtering calls and managing of filter state in URL.

In the header, there is a button that opens up the filter editing popup. When there are filters, we will dynamically determine what to show based on the available width:

No filters:
Screenshot 2024-08-12 at 10 47 44 AM

There are filters but none can be shown in the available space:
Screenshot 2024-08-12 at 10 47 59 AM

All filters fit in the available space
Screenshot 2024-08-12 at 10 50 10 AM

Too many filters to show in the available space, but some fit
Screenshot 2024-08-12 at 10 48 43 AM

If there's room to show a filter tag, user can one-click remove it with the 'x' button on the tag.

Clicking on the button opens up a dialog:

Screenshot 2024-08-12 at 10 55 40 AM

Each row corresponds to one filter. The 'x' button can be used to remove a filter, or you can remove all of them at once.
The available operators depend on the selected field, and the type of value input depends on the operator

Column selection:
Screenshot 2024-08-12 at 10 55 18 AM

Operator selection:
Screenshot 2024-08-12 at 10 55 12 AM

### CODE REVIEW

File: weave-js/src/components/PagePanelComponents/Home/Browse3/filters/CellFilterWrapper.tsx [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  children,
+  onAddFilter,
+  field,
+  operation,
+  value,
+}: CellFilterWrapperProps) => {
+  const onClickCapture = onAddFilter
+    ? (e: React.MouseEvent) => {
+        // event.altKey - pressed Option key on Macs
+        if (e.altKey) {
```
Comment (tssweeney) [inline weave-js/src/components/PagePanelComponents/Home/Browse3/filters/CellFilterWrapper.tsx:26]: I could see a nice little "did you know? ... " footer on the filter modal telling users the can option-click

File: weave-js/src/components/PagePanelComponents/Home/Browse3/filters/ValueInputBoolean.tsx [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  onSetValue: (value: string) => void;
+};
+
+export const ValueInputBoolean = ({
+  value,
+  onSetValue,
+}: ValueInputBooleanProps) => {
+  const isTrue = value === 'true' || value === true;
+  const isFalse = value === 'false' || value === false;
+  const classNamesBtn = classNames(
```
Comment (tssweeney) [inline weave-js/src/components/PagePanelComponents/Home/Browse3/filters/ValueInputBoolean.tsx:18]: are these constants? Maybe pull them out?

File: weave-js/src/components/PagePanelComponents/Home/Browse3/filters/types.ts [hunk 3]
```diff
@@ -0,0 +1,250 @@
...
+  GridFilterModel,
+} from '@mui/x-data-grid-pro';
+import _ from 'lodash';
+
+import {TraceCallSchema} from '../pages/wfReactInterface/traceServerClientTypes';
+
+export type FilterId = number | string | undefined;
+
+// These are columns we won't allow the user to filter on.
+// For most of these it would be great if we could enable filtering in the future.
```
Comment (tssweeney) [inline weave-js/src/components/PagePanelComponents/Home/Browse3/filters/types.ts:14]: Ack - we definitely will add support for these

File: weave-js/src/components/PagePanelComponents/Home/Browse3/filters/types.ts [hunk 4]
```diff
@@ -0,0 +1,250 @@
...
+};
+
+export const FIELD_TYPE: Record<string, string> = {
+  id: 'id',
+  'summary.weave.status': 'status',
+  wb_user_id: 'user',
+  started_at: 'datetime',
+};
+
+export const getFieldType = (field: string): string => {
```
Comment (tssweeney) [inline weave-js/src/components/PagePanelComponents/Home/Browse3/filters/types.ts:47]: this file seems to have a lot of overlap with the `operators.ts` file - any use in sharing some code?

**Review by tssweeney — APPROVED** [review]
This looks great! I added a few comments, but everything looks pretty well factored. Clicked around in the UI - seems really nice!

File: weave-js/src/components/PagePanelComponents/Home/Browse3/filters/CellFilterWrapper.tsx (hunk 1 above)
Comment (jamie-rasmussen) [reply weave-js/src/components/PagePanelComponents/Home/Browse3/filters/CellFilterWrapper.tsx:26]: I think this is a good idea, let me follow up with design. Might hold off until it works for status/costs/tokens too.

File: weave-js/src/components/PagePanelComponents/Home/Browse3/filters/types.ts (hunk 4 above)
Comment (jamie-rasmussen) [reply weave-js/src/components/PagePanelComponents/Home/Browse3/filters/types.ts:47]: You're right, I actually think we can get rid of a lot of operators.ts with this PR.

**Pull request merged**